pynicotine/portmapper.py

```diff
@@ -126,17 +127,25 @@ def _request_port_mapping(self, public_port, private_port, lease_duration):
 
         with socket.socket(family=socket.AF_INET, type=socket.SOCK_DGRAM, proto=socket.IPPROTO_UDP) as sock:
             sock.bind((self.local_ip_address, 0))
-            sock.settimeout(self.REQUEST_TIMEOUT)
 
             request = self.PortmapRequest(public_port, private_port, lease_duration)
-            request.sendto(sock, (self._gateway_address, self.REQUEST_PORT))
 
-            try:
-                response = self.PortmapResponse(message=sock.recv(16))
-                return response.result
+            timeout = self.REQUEST_INIT_TIMEOUT
+            i = 1
+            while i <= self.REQUEST_ATTEMPTS:
```

Comment:
suggest `for num_attempt in range(self.REQUEST_ATTEMPTS):` would eliminate the need to define `i`.

Also, an empty line is desired above the block.